pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -218,11 +231,14 @@ private PulsarClientImpl(ClientConfigurationData conf, EventLoopGroup eventLoopG
                     new ExecutorProvider(conf.getNumListenerThreads(), "pulsar-external-listener");
             this.internalExecutorProvider = internalExecutorProvider != null ? internalExecutorProvider :
                     new ExecutorProvider(conf.getNumIoThreads(), "pulsar-client-internal");
+            this.lookupExecutorProvider = lookupExecutorProvider != null ? lookupExecutorProvider :
+                    new ExecutorProvider(conf.getNumIoThreads(), "pulsar-client-lookup");
```

Comment:
I don't think that numIoThread should be used for this purpose. It's already used for the `scheduledExecutorProvider`, but that's not a great default at all. This could cause regressions due to memory increase.

Reply:
Is it better to use a single thread?

Reply:
yes, it's better than numIoThreads.

Reply:
@lhotari Done.